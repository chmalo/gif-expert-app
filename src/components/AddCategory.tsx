import {
  ChangeEvent,
  Dispatch,
  FormEvent,
  SetStateAction,
  useState,
} from "react"

const AddCategory = ({
  setCategories,
}: {
  setCategories: Dispatch<SetStateAction<string[]>>
}) => {
  const [inputValue, setInputValeu] = useState("")

  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
    setInputValeu(e.target.value)
  }
  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault()

    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats])
      setInputValeu("")
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  )
}

export default AddCategory
