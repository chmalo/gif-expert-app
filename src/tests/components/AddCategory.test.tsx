import { shallow } from "enzyme"
import { beforeEach, describe, expect, test, vi } from "vitest"
import AddCategory from "../../components/AddCategory"

describe("Testing AddCategory", () => {
  const setCategories = vi.fn(() => {})
  let wrapper = shallow(<AddCategory setCategories={setCategories} />)

  beforeEach(() => {
    vi.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })

  test("it should show correctly ", () => {
    expect(wrapper).toMatchSnapshot()
  })

  test("it should change the text box", () => {
    const input = wrapper.find("input")
    const value = "hola christian"
    input.simulate("change", { target: { value } })

    expect(wrapper.find("p").text().trim()).toBe(value)
  })

  test("it should not the information with submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault: () => {} })

    expect(setCategories).not.toHaveBeenCalled()
  })

  test("it should call setcategories and clear the text box", () => {
    const input = wrapper.find("input")
    const value = "hola christian"
    input.simulate("change", { target: { value } })

    const form = wrapper.find("form")
    form.simulate("submit", { preventDefault: () => {} })

    expect(setCategories).toHaveBeenCalled()
    expect(setCategories).toHaveBeenCalledTimes(1)
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function))

    expect(input.prop("value")).toBe("")
  })
})
