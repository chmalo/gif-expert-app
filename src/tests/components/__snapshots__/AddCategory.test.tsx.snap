// Vitest Snapshot v1

exports[`Testing AddCategory > it should show correctly  1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
